lib/cn/cn_tree.c

```diff
@@ -2220,6 +2460,23 @@ cn_comp_compact(struct cn_compaction_work *w)
     hp = w->cw_tree->ct_kvdb_health;
     assert(hp);
 
+    if (w->cw_rule == CN_RULE_ZSPILL) {
+        bool zspilled;
+
+        w->cw_horizon = cn_get_seqno_horizon(w->cw_tree->cn);
+        w->cw_cancel_request = cn_get_cancel(w->cw_tree->cn);
+
+        perfc_inc(w->cw_pc, PERFC_BA_CNCOMP_START);
+
+        cn_setname(w->cw_threadname);
+
+        w->cw_err = cn_zspill(w, &zspilled);
+        if (w->cw_err || zspilled)
+            return;
+
+        w->cw_rule = CN_RULE_RSPILL;
+    }
+
```

Comment:
Would rather you not use rule for this purpose, why not use action?

Reply:
Ok, I'd at least check that action is a CN_ACTION_SPILL, then check the rule.

Reply:
Looking at the changes to cn_comp_compact() we can now more easily support a CN_ACTION_ZSPILL and avoid keying on rule.  Rule was meant to be an advisory/informative kind of thing, but this patch now requires it to be one specific code (e.g., I should be able to support several rules for zspill, but with this approach that's not possible without an explosion of checks to support it).

Reply:
Yes, good point. Fixed it to now key on action alone